core/lib/types/src/storage/writes/compression.rs

```diff
@@ -0,0 +1,409 @@
+use super::*;
+
+// Starting with version 1 for this compression strategy. Any modifications to our current strategy MUST
+// increment this number.
+pub const COMPRESSION_VERSION_NUMBER: u8 = 1;
+
+// Trait used to define functionality for different compression modes. Defines functions for
+// output size, what type of operation was performed, and value/extended compression.
+trait CompressionMode: 'static {
+    /// Id of the operation being performed.
+    fn operation_id(&self) -> usize;
+    /// Gets the diff and size of value
+    fn get_diff_and_size(&self) -> Option<(U256, usize)>;
+    /// Number of bytes the compressed value requires. None indicates that compression cannot be performed for the
+    /// given strategy.
+    fn output_size(&self) -> Option<usize> {
+        self.get_diff_and_size().map(|(_, size)| size)
+    }
+    /// Compress the value.
+    fn compress_value_only(&self) -> Option<Vec<u8>> {
+        let (diff, size) = self.get_diff_and_size()?;
+
+        let mut buffer = [0u8; 32];
+        diff.to_big_endian(&mut buffer);
+
+        let diff = buffer[(32 - size)..].to_vec();
+
+        Some(diff)
+    }
+    /// Concatenation of the metadata byte (5 bits for len and 3 bits for operation type) and the compressed value.
+    fn compress_extended(&self) -> Option<Vec<u8>> {
+        self.compress_value_only().map(|compressed_value| {
+            let mut res: Vec<u8> = vec![];
+            res.push(metadata_byte(
+                self.output_size().unwrap(),
+                self.operation_id(),
+            ));
+            res.extend(compressed_value);
+            res
+        })
+    }
+}
+
+struct CompressionByteAdd {
+    pub prev_value: U256,
+    pub new_value: U256,
+}
+
+impl CompressionMode for CompressionByteAdd {
+    fn operation_id(&self) -> usize {
+        1
+    }
+
+    fn get_diff_and_size(&self) -> Option<(U256, usize)> {
+        let diff = self.new_value.overflowing_sub(self.prev_value).0;
+        // Ceiling division
+        let size = (diff.bits() + 7) / 8;
+
+        if size >= 31 {
+            None
+        } else {
+            Some((diff, size))
+        }
+    }
+
+    fn output_size(&self) -> Option<usize> {
+        self.get_diff_and_size().map(|(_, size)| size)
+    }
+
+    fn compress_value_only(&self) -> Option<Vec<u8>> {
+        let (diff, size) = self.get_diff_and_size()?;
+
+        let mut buffer = [0u8; 32];
+        diff.to_big_endian(&mut buffer);
+
+        let diff = buffer[(32 - size)..].to_vec();
+
+        Some(diff)
+    }
+
+    fn compress_extended(&self) -> Option<Vec<u8>> {
```

Comment:
No need to overwrite `compress_extended` here

Reply:
@koloz193 will you apply this?

Reply:
yup!

Reply:
just pushed an update with it

Reply:
hmm, I don't see it

Reply:
sorry `zk fmt` issues, should be reflected now